framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -76,6 +80,32 @@ protected void commonAssignmentCheck(
                     getTypeFactory().createIntRangeAnnotation(Range.CHAR_EVERYTHING));
         }
 
+        if (valueTree.getKind() == Kind.METHOD_INVOCATION
+                && TreeUtils.isArrayscopyOfMethodInvocation((MethodInvocationTree) valueTree)
+                && valueType.getKind() == TypeKind.ARRAY) {
+            List<? extends ExpressionTree> args = ((MethodInvocationTree) valueTree).getArguments();
+            if (getTypeFactory().getAnnotatedType(args.get(1)).getAnnotation(IntVal.class)
```

Comment:
Rather than checking for `IntVal` and `IntRange` by hand, I suggest using `ValueCheckerUtils.getPossibleValues`, which returns a `Range` or `null` if the annotation isn't integral. It already has support for both `IntVal` and `IntRange`, and you should be able to just use its lower bound directly rather than doing the computations below. That will greatly simplify the code.